### PR TITLE
EVAL_RO & EVALSHA_RO commands

### PR DESCRIPTION
**Pull request by shacharPash**

### CODE REVIEW

**Comment by shacharPash** [comment]
Work in progress.

File: src/StackExchange.Redis/Interfaces/IDatabase.cs [hunk 1]
```diff
@@ -1298,6 +1298,31 @@ public interface IDatabase : IRedis, IDatabaseAsync
...
+        /// <param name="script">The script to execute.</param>
+        /// <param name="keys">The keys to execute against.</param>
+        /// <param name="values">The values to execute against.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>A dynamic representation of the script's result.</returns>
+        /// <remarks>
+        /// <seealso href="https://redis.io/commands/eval_ro"/>,
+        /// <seealso href="https://redis.io/commands/evalsha_ro"/>
+        /// </remarks>
+        RedisResult ScriptEvaluateRO(string script, RedisKey[]? keys = null, RedisValue[]? values = null, CommandFlags flags = CommandFlags.None);
```
Comment (mgravell) [inline src/StackExchange.Redis/Interfaces/IDatabase.cs:1313]: naming: we've usually tried to be explicit in the API, to trade brevity for clarity; if this RO is "read only", I suggest `ScriptEvaluateReadOnly`, although I almost wonder whether a `CommandFlags` makes sense (probably not)

File: src/StackExchange.Redis/RedisDatabase.cs [hunk 2]
```diff
@@ -1525,9 +1511,24 @@ public Task<RedisResult> ExecuteAsync(string command, ICollection<object>? args,
             return ExecuteAsync(msg, ResultProcessor.ScriptResult, defaultValue: RedisResult.NullSingle);
         }
 
+        public RedisResult ScriptEvaluate(string script, RedisKey[]? keys = null, RedisValue[]? values = null, CommandFlags flags = CommandFlags.None)
+        {
+            var command = ResultProcessor.ScriptLoadProcessor.IsSHA1(script) ? RedisCommand.EVALSHA : RedisCommand.EVAL;
```
[the remark this answers is not in the document]
Comment (shacharPash) [reply src/StackExchange.Redis/RedisDatabase.cs:1516]: I also do not like it that way, I tried to optimize this line but the problem is that the lines are not the same, in some of the lines you have to place the commands "EVAL/EVALSHA" and in some the commands "EVAL_RO/EVALSHA_RO"

I didn't understand your I did not understand your suggestion to use private helper API that takes bool readOnly, If this is still relevant despite what I have written I would like for you to elaborate further.

**Comment by shacharPash** [comment]
There is a lot of errors in the tests file, I guess the problem is with my computer so I try to fix it before I write the tests to the commands I added, while you can review the code.

**Comment by chayim** [comment]
@shacharPash This may have a better change of being addressed if there were unit tests. WDYT about contributing some?

**Comment by shacharPash** [comment]
Hey @mgravell , I added some tests to the commands in this PR, I would appreciate it if you would take a look.

**Comment by NickCraver** [comment]
@shacharPash no ask at the moment, @mgravell is going to take a look at if this API surface area makes sense (given it's a subset of existing APIs which are a bit fractured)

**Review by mgravell — APPROVED** [review]

**Pull request merged**